autoscaler/controllers/actions/sampling_controller.go

```diff
@@ -0,0 +1,254 @@
+			OrderHint:       -24,
+			ProcessorConfig: runtime.RawExtension{Raw: samplingConfigJson},
+		},
+	}
+
+	groupByTraceProcessor := r.getGroupByTraceProcessor(namespace, actionsReferences)
+	if err := r.Patch(ctx, groupByTraceProcessor, client.Apply, client.FieldOwner("groupbytrace"), client.ForceOwnership); err != nil {
+		return err
+	}
+
+	return r.Patch(ctx, samplingProcessor, client.Apply, client.FieldOwner("sampling-processor"), client.ForceOwnership)
+}
+
+func (r *OdigosSamplingReconciler) ReportReconciledToProcessorFailed(ctx context.Context, action metav1.Object, reason, msg string) error {
+	conditions, err := getConditions(action)
+	if err != nil {
+		return err
+	}
+
+	changed := meta.SetStatusCondition(conditions, metav1.Condition{
+		Type:               "ActionTransformedToProcessorType",
+		Status:             metav1.ConditionFalse,
+		Reason:             reason,
+		Message:            msg,
+		ObservedGeneration: action.GetGeneration(),
+	})
+
+	if changed {
+		return r.Status().Update(ctx, action.(client.Object))
+	}
+	return nil
+}
+
+func (r *OdigosSamplingReconciler) ReportReconciledToProcessor(ctx context.Context, action metav1.Object) error {
+	conditions, err := getConditions(action)
+	if err != nil {
+		return err
+	}
+
+	changed := meta.SetStatusCondition(conditions, metav1.Condition{
+		Type:               "ActionTransformedToProcessorType",
+		Status:             metav1.ConditionTrue,
+		Reason:             "ProcessorCreatedReason",
+		Message:            "The action has been reconciled to a processor resource.",
+		ObservedGeneration: action.GetGeneration(),
+	})
+
+	if changed {
+		if err := r.Status().Update(ctx, action.(client.Object)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *OdigosSamplingReconciler) filterActions(ctx context.Context, actions []metav1.Object, handler sampling.ActionHandler) []metav1.Object {
+	logger := log.FromContext(ctx)
+
+	var filteredActions []metav1.Object
+	for _, action := range actions {
+
+		// filter disabled actions
+		if handler.IsActionDisabled(action) {
+			continue
+		}
+		// filter actions with invalid configuration
+		if err := handler.ValidateRuleConfig(handler.GetRuleConfig(action)); err != nil {
+			logger.V(0).Error(err, "Failed to validate rule config")
+			r.ReportReconciledToProcessorFailed(context.Background(), action, "FailedToTransformToProcessorReason", err.Error())
+			continue
+		}
+
+		filteredActions = append(filteredActions, action)
+	}
+	return filteredActions
+}
+
+func (r *OdigosSamplingReconciler) isRelevantActions(desiredActions map[reflect.Type][]metav1.Object) bool {
+	for _, actions := range desiredActions {
+		if len(actions) > 0 {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *OdigosSamplingReconciler) deleteSamplingProcessorsIfExists(ctx context.Context, namespace string) error {
+	if err := commonproc.DeleteProcessorByType(ctx, r.Client, GroupByTraceType, namespace); err != nil {
```

Comment:
I wonder if we can delete the processor by it's known name ("groupbytrace-processor") which is direct, instead of using `List` and then iterating all the items. might be a bit simpler